### PR TITLE
Introducing support to debug.SetGCPercent.

### LINKED ISSUE #8: Support runtime.SetGCPercent
Provide a command similar to runtime.SetGCPercent to change the GC percentage dynamically during runtime.

```
$ gops setgc -p=<pid> 60 # Sets GC to 60%
$ gops setgc -p=<pid> -1 # Disables GC
```

### PR DESCRIPTION
**Pull request by danielfireman**
On my I needed to augment a little bit the communication protocol to add support to optional parameters. Fixes #8

### CODE REVIEW

**Comment by danielfireman** [comment]
I signed it!

Em 5 de out de 2017 21:55, "googlebot" <notifications@github.com> escreveu:

File: cmd.go [hunk 1]
```diff
@@ -138,12 +153,12 @@ func pprof(addr net.TCPAddr, p byte) error {
...
 }
 
-func stats(addr net.TCPAddr) error {
+func stats(addr net.TCPAddr, _ []string) error {
 	return cmdWithPrint(addr, signal.Stats)
 }
 
-func cmdWithPrint(addr net.TCPAddr, c byte) error {
-	out, err := cmd(addr, c)
+func cmdWithPrint(addr net.TCPAddr, c byte, p ...byte) error {
```
Comment (rakyll) [inline cmd.go]: params ...byte

keep it consistent.

File: agent/agent.go [hunk 2]
```diff
@@ -230,6 +231,12 @@ func handle(conn io.Writer, msg []byte) error {
 		trace.Start(conn)
 		time.Sleep(5 * time.Second)
 		trace.Stop()
+	case signal.SetGCPercent:
+		perc, err := binary.ReadVarint(bufio.NewReader(conn))
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(conn, "New GC percent set to %v. Previous value %v \n", perc, debug.SetGCPercent(int(perc)))
```
[the remark this answers is not in the document]
Comment (danielfireman) [reply agent/agent.go]: Done.

File: cmd.go [hunk 3]
```diff
@@ -28,35 +29,49 @@ var cmds = map[string](func(addr net.TCPAddr) error){
 	"pprof-cpu":  pprofCPU,
 	"stats":      stats,
 	"trace":      trace,
+	"setgc":      setGC,
 }
 
-func stackTrace(addr net.TCPAddr) error {
+func setGC(addr net.TCPAddr, p []string) error {
+	if len(p) != 1 {
+		fmt.Println("Missing foo")
```
[the remark this answers is not in the document]
Comment (danielfireman) [reply cmd.go]: Done.

File: cmd.go (hunk 1 above)
Comment (danielfireman) [reply cmd.go]: Done.

**Comment by danielfireman** [comment]
All done. Please take another look.

**Comment by rakyll** [comment]
Neat! We also need to update the README.

**Comment by danielfireman** [comment]
Totally. I'll send a PR with the Readme update.

**Comment by rakyll** [comment]
Sent https://github.com/google/gops/pull/57 before seeing this, sorry @danielfireman.

**Comment by danielfireman** [comment]
Sure! Thanks!

Em 8 de out de 2017 16:29, "JBD" <notifications@github.com> escreveu:

Sent #57 <https://github.com/google/gops/pull/57> before seeing this, sorry
@danielfireman <https://github.com/danielfireman>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/google/gops/pull/56#issuecomment-335031727>, or mute
the thread

**Pull request merged**